fastplotlib/graphics/selectors/_rectangle.py

```diff
@@ -0,0 +1,511 @@
+            bottom_left_vertex,
+            bottom_right_vertex,
+            top_left_vertex,
+            top_right_vertex,
+        )
+
+        for vertex in self.vertices:
+            vertex.world.z = -0.25
+            group.add(vertex)
+
+        self._selection = RectangleSelectionFeature(
+            selection, axis=axis, limits=self._limits
+        )
+
+        # include parent offset
+        if parent is not None:
+            offset = (parent.offset[0], parent.offset[1], 0)
+        else:
+            offset = (0, 0, 0)
+
+        BaseSelector.__init__(
+            self,
+            edges=self.edges,
+            fill=(self.fill,),
+            vertices=self.vertices,
+            hover_responsive=(*self.edges, *self.vertices),
+            arrow_keys_modifier=arrow_keys_modifier,
+            axis=axis,
+            parent=parent,
+            name=name,
+            offset=offset,
+        )
+
+        self._set_world_object(group)
+
+        self.selection = selection
+
+    def get_selected_data(
+        self, graphic: Graphic = None, mode: str = "full"
+    ) -> Union[np.ndarray, List[np.ndarray]]:
+        """
+        Get the ``Graphic`` data bounded by the current selection.
+        Returns a view of the data array.
+
+        If the ``Graphic`` is a collection, such as a ``LineStack``, it returns a list of views of the full array.
+        Can be performed on the ``parent`` Graphic or on another graphic by passing to the ``graphic`` arg.
+
+        Parameters
+        ----------
+        graphic: Graphic, optional, default ``None``
+            if provided, returns the data selection from this graphic instead of the graphic set as ``parent``
+        mode: str, default 'full'
+            One of 'full', 'partial', or 'ignore'. Indicates how selected data should be returned based on the
+            selectors position over the graphic. Only used for ``LineGraphic``, ``LineCollection``, and ``LineStack``
+            If 'full', will return all data bounded by the x and y limits of the selector even if partial indices
+            alone one axis are not fully covered by the selector.
+            If 'partial' will return only the data that is bounded within the limits.
+            If 'ignore', will not return data if selector is not fully covering the graphic along the axes bounds.
+
+        Returns
+        -------
+        np.ndarray or List[np.ndarray]
+            view or list of views of the full array, returns ``None`` if selection is empty
+        """
+        source = self._get_source(graphic)
+        ixs = self.get_selected_indices(source)
+
+        # do not need to check for mode for images, because the selector is bounded by the image shape
+        # will always be `full`
+        if "Image" in source.__class__.__name__:
+            s_x = slice(ixs[0][0], ixs[0][-1] + 1)
+            s_y = slice(ixs[1][0], ixs[1][-1] + 1)
+
+            return source.data[s_x, s_y]
```

Comment:
```suggestion
            col_ixs = slice(ixs[0][0], ixs[0][-1] + 1)
            row_ixs = slice(ixs[1][0], ixs[1][-1] + 1)

            return source.data[row_ixs, col_ixs]
```

Depends on a suggested change below